packages/stargate/src/signingstargateclient.ts

```diff
@@ -198,6 +206,44 @@ export class SigningStargateClient extends StargateClient {
     return this.signAndBroadcast(senderAddress, [sendMsg], this.fees.send, memo);
   }
 
+  public async delegateTokens(
+    senderAddress: string,
+    validatorAddress: string,
+    amount: Coin,
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const delegateMsg = {
+      typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
+      value: MsgDelegate.fromPartial({ delegatorAddress: senderAddress, validatorAddress, amount }),
+    };
```

Comment:
Is there a type-safe way to create messages like this? I mean binding the URL to the value type? Bit I guess we'd have to create interfaces for each message type for this. Any thoughts?

Reply:
We could do something like this?
```ts
type RegistryMap = {
  "/cosmos.bank.v1beta1.MsgMultiSend": MsgMultiSend;
  "/cosmos.distribution.v1beta1.MsgFundCommunityPool": MsgFundCommunityPool;
  // ...
};

interface NotJustAny<K extends keyof RegistryMap> {
  typeUrl: K;
  value: RegistryMap[K];
}
```

Reply:
Is this better than a decentral type system like this?

```ts
interface EncodeMsgDelegate extends EncodeObject {
  readonly typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
  readonly value: MsgDelegate,
}
```

Reply:
Not really. Do we want this overhead?

Reply:
We have code like this a bunch of times already:

```ts
        const msg = MsgDelegate.fromPartial({
          delegatorAddress: faucet.address0,
          validatorAddress: validator.validatorAddress,
          amount: coin(1234, "ustake"),
        });
        const msgAny = {
          typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
          value: msg,
        };
```

I'm a bit concerned about typos in those type URLs or URL/type mismatches.

But we can leave it like it is right now. Don't have a strong opinion on this.

Reply:
Type safety is always nice: https://github.com/cosmos/cosmjs/issues/734